test/flash.test.js

```diff
@@ -739,3 +740,64 @@ QUnit.test('fires loadedmetadata after first segment append', function() {
   timers.runAll();
   QUnit.equal(loadedmetadataCount, 1, 'loadedmetadata does not fire after second append');
 });
+
+QUnit.test('cleans up WebVTT cues on hls dispose', function() {
+  let sourceBuffer = this.mediaSource.addSourceBuffer('video/mp2t');
+
+  let addedTracks = [];
+  let removedTracks = [];
+  let metadata = [{
+    cueTime: 2,
+    frames: [{
+      url: 'This is a url tag'
+    }, {
+      value: 'This is a text tag'
+    }]
+  }, {
+    cueTime: 12,
+    frames: [{
+      data: 'This is a priv tag'
+    }]
+  }];
+  let captions = [{
+    startTime: 1,
+    endTime: 3,
+    text: 'This is an in-band caption'
+  }];
+
+  metadata.dispatchType = 0x10;
+  this.mediaSource.player_ = {
+    addRemoteTextTrack(options) {
+      let trackEl = {
+        track: {
+          kind: options.kind,
+          label: options.label,
+          addCue(cue) {}
+        }
+      };
+
+      addedTracks.push(trackEl.track);
+      return trackEl;
+    },
+    remoteTextTracks() {
+      return addedTracks;
+    },
+    removeRemoteTextTrack(track) {
+      removedTracks.push(track.kind);
+    }
+  };
+
+  sourceBuffer.segmentParser_.trigger('data', {
+    tags: {
+      videoTags: [],
+      audioTags: []
+    },
+    metadata,
+    captions
+  });
+  sourceBuffer.segmentParser_.trigger('done');
+
+  QUnit.equal(addedTracks.length, 2, 'created two remote TextTracks');
```

Comment:
We have tests that catch this scenario already and make sure the proper tracks are created for captions and metadata individually but it is a good idea to check that here for completeness.